api/inc/uvisor_exports.h

```diff
@@ -78,10 +78,15 @@
  *
  * This macro works from both inside and outside function scope.
  *
- * FIXME This is currently not implemented. This issue is tracked at
- * https://github.com/ARMmbed/uvisor/issues/288
+ * The implementations differ due to compilation differences, c++ static_assert
+ * is known from c++11 while mbed-os compiles with c++98
  */
-#define UVISOR_STATIC_ASSERT(cond, msg)
+#ifdef  __cplusplus
+#define UVISOR_STATIC_ASSERT(cond, msg) typedef char STATIC_ASSERT_##msg[(cond)?1:-1]
+#else
+#define UVISOR_STATIC_ASSERT(cond, msg) _Static_assert(cond, #msg)
+#endif/*__CPP__*/
```

Comment:
Remove `/*__CPP__*/` comment. It's annoying to maintain and doesn't help much, since the ifdef is so small.